components/expenses/TaxFormLinkModal.js

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { gql, useMutation } from '@apollo/client';
+import { get } from 'lodash';
+import { FormattedMessage } from 'react-intl';
+
+import { API_V2_CONTEXT } from '../../lib/graphql/helpers';
+
+import Container from '../Container';
+import { Box } from '../Grid';
+import MessageBox from '../MessageBox';
+import MessageBoxGraphqlError from '../MessageBoxGraphqlError';
+import StyledButton from '../StyledButton';
+import StyledInput from '../StyledInput';
+import StyledInputField from '../StyledInputField';
+import StyledLink from '../StyledLink';
+import StyledSelect from '../StyledSelect';
+import { H2, Span } from '../Text';
+import { TOAST_TYPE, useToasts } from '../ToastProvider';
+
+import StyledModal, { ModalBody, ModalHeader } from './../StyledModal';
+
+const setTaxFormMutation = gql`
+  mutation SetTaxForm($account: AccountReferenceInput!, $taxFormLink: URL!, $year: Int!) {
+    setTaxForm(account: $account, taxFormLink: $taxFormLink, year: $year) {
+      success
+    }
+  }
+`;
+
+const arrayOfYears = () => {
+  const max = new Date().getFullYear();
+  const min = 2018;
+  const years = [];
+
+  for (let i = max; i >= min; i--) {
+    years.push(i);
+  }
+  return years;
+};
+
+const TaxFormLinkModal = ({ account, year, onClose, expenseData }) => {
+  const [taxFormLink, setTaxFormLink] = React.useState(null);
+  const taxYearOptions = arrayOfYears().map(year => ({ key: year, value: year, label: year }));
+  const defaultTaxYearOption = taxYearOptions.filter(option => option.value === year)[0];
+  const [taxFormYear, setTaxFormYear] = React.useState(defaultTaxYearOption.value);
+  const { addToast } = useToasts();
+  const [error, setError] = React.useState(null);
+  const [submit, { loading }] = useMutation(setTaxFormMutation, { context: API_V2_CONTEXT });
+  return (
+    <StyledModal role="alertdialog" width="578px" onClose={onClose} trapFocus>
+      <ModalHeader>
+        <H2 mb={2} fontSize={'28px'}>
+          <FormattedMessage defaultMessage="Tax Form for {accountName}" values={{ accountName: account.name }} />
+        </H2>
+      </ModalHeader>
+      <ModalBody>
+        <MessageBox type="info" withIcon>
+          <FormattedMessage
+            defaultMessage="Please upload the tax form to the {linkToTaxFormsFolder}. Copy the link to the file and paste it in the field below."
+            values={{
+              linkToTaxFormsFolder: (
+                <StyledLink
+                  href="https://drive.google.com/drive/folders/1ga_-6tBTqADvngRr9nbxRKf7yoSne1n1?usp=share_link"
+                  openInNewTab
+                >
+                  Google Drive Folder
+                </StyledLink>
+              ),
+            }}
+          />
```

Comment:
Great that you added this message; admins often don't know where to upload docs. Also Fyi, there's this unresolved situation: https://opencollective.slack.com/archives/GST5GVCQ7/p1674030322385029?thread_ts=1673992250.622439&cid=GST5GVCQ7.

Small change requested on the translation itself as it's incorrect, see https://docs.opencollective.com/help/contributing/development/translations#translate-links-inline.

```suggestion
          <FormattedMessage
            defaultMessage="Please upload the tax form to the <Link>Google Drive Folder</Link>. Copy the link to the file and paste it in the field below."
            values={{
              linkToTaxFormsFolder: getI18nLink({ href: 'https://drive.google.com/drive/folders/1ga_-6tBTqADvngRr9nbxRKf7yoSne1n1?usp=share_link', openInNewTab: true }}
          />
```

Reply:
@Betree : I've done the translation change. 


> Great that you added this message; admins often don't know where to upload docs. Also Fyi, there's this unresolved situation: https://opencollective.slack.com/archives/GST5GVCQ7/p1674030322385029?thread_ts=1673992250.622439&cid=GST5GVCQ7.

The link here seems broken. Could you please tell me what this is about?